action/protocol/staking/candidate_statemanager.go

```diff
@@ -38,8 +38,8 @@ type (
 		GetByOwner(address.Address) *Candidate
 		GetBySelfStakingIndex(uint64) *Candidate
 		Upsert(*Candidate) error
-		CreditBucketPool(*big.Int, bool) error
-		DebitBucketPool(*big.Int, bool, bool) error
+		CreditBucketPool(*big.Int) error
+		DebitBucketPool(*big.Int, bool) error
```

Comment:
think we should pass in ctx, and use ctx.GreenlandHeight inside bucketPool to determine if create or not

Reply:
similar to how the rewarding pool PR handles this

Reply:
ctx has been passed in to csm via NewCandidateStateManager. One csm is supposed to be used throughout the live of a ctx.